OSDOCS#12987: Two-node OpenShift cluster with fencing (Technology Preview) (#98503)
openshift/openshift-docs: installing/installing_two_node_cluster/installing_tnf/installing-two-node-fencing.adoc

:_mod-docs-content-type: ASSEMBLY
[id="installing-two-node-fencing_{context}"]
= Preparing to install a two-node OpenShift cluster with fencing (Technology Preview)
include::_attributes/common-attributes.adoc[]
:context: installing-two-node-fencing

toc::[]

:FeatureName: Two-node OpenShift cluster with fencing
include::snippets/technology-preview.adoc[leveloffset=+1]

A two-node OpenShift cluster with fencing provides high availability (HA) with a reduced hardware footprint. This configuration is designed for distributed or edge environments where deploying a full three-node control plane cluster is not practical.

A two-node cluster does not include compute nodes. The two control plane machines run user workloads in addition to managing the cluster.

> Fencing is managed by Pacemaker, which can isolate an unresponsive node by using the node’s Baseboard Management Console (BMC). After the unresponsive node is fenced, the remaining node can safely continue operating the cluster without the risk of resource corruption.

Reviewer: The IBM SG  has a gripe about stating object's in a possessive way.

How about:

> Fencing is managed by Pacemaker, which can isolate an unresponsive node by using the Baseboard Management Console (BMC) of the node.

[NOTE]
====

> You can deploy a two-node OpenShift cluster with fencing by using either the user-provisioned infrastructure (UPI) method or the installer-provisioned infrastructure (IPI) method. A two-node deployment path by using the Assisted Installer is planned for a future release.

Reviewer: ```suggestion
You can deploy a two-node OpenShift cluster with fencing by using either the user-provisioned infrastructure method or the installer-provisioned infrastructure method. A two-node deployment path by using the Assisted Installer is planned for a future release.
```

Author: I can remove it for now instead of documenting something that is not supported.

====

The two-node OpenShift cluster with fencing requires the following hosts:

.Minimum required hosts
[options="header"]
|===

|Hosts |Description

|Two control plane machines
|The control plane machines run the Kubernetes and {product-title} services that form the control plane.

|One temporary bootstrap machine
|You need a bootstrap machine to deploy the {product-title} cluster on the control plane machines. You can remove the bootstrap machine after you install the cluster.

|===

The bootstrap and control plane machines must use Red Hat Enterprise Linux CoreOS (RHCOS) as the operating system. For instructions on installing RHCOS and starting the bootstrap process, see xref:../../../installing/installing_bare_metal/upi/installing-bare-metal#creating-machines-bare-metal_installing-bare-metal[Installing RHCOS and starting the {product-title} bootstrap process]
[NOTE]
====
The requirement to use RHCOS applies only to user-provisioned infrastructure deployments. For installer-provisioned infrastructure deployments, the bootstrap and control plane machines are provisioned automatically by the installation program, and you do not need to manually install RHCOS.

> ====

Reviewer 2: ```suggestion
The bootstrap and control plane machines must use Red Hat Enterprise Linux CoreOS (RHCOS) as the operating system. For instructions on installing RHCOS and starting the bootstrap process, see xref:../../../installing/installing_bare_metal/upi/installing-bare-metal#creating-machines-bare-metal_installing-bare-metal[Installing RHCOS and starting the {product-title} bootstrap process]

[NOTE]
====
The requirement to use RHCOS applies only to user-provisioned infrastructure deployments. For installer-provisioned infrastructure deployments, the bootstrap and control plane machines are provisioned automatically by the installation program, and you do not need to manually install RHCOS.
====
```

Reviewer 2: Fix in follow-up: https://github.com/openshift/openshift-docs/pull/100785

> This requirement applies only to UPI deployments. For IPI deployments, the bootstrap and control plane machines are provisioned automatically by the installer, and you do not need to manually install RHCOS.

Reviewer: ```suggestion
The requirement to use RHCOS applies only to user-provisioned infrastructure deployments. For IPI deployments, the bootstrap and control plane machines are provisioned automatically by the installer, and you do not need to manually install RHCOS.
```

include::modules/installation-two-node-cluster-min-resource-reqs.adoc[leveloffset=+1]

// Two-node-dns-requirements - UPI
include::modules/installation-dns-user-infra.adoc[leveloffset=+1]

// Two-node-dns-requirements - IPI
include::modules/installation-dns-installer-infra.adoc[leveloffset=+1]

// Cofiguration for Ingress LB to get it to work with Pacemaker
include::modules/installation-two-node-ingress-lb-configuration.adoc[leveloffset=+1]

//  Creating a manifest object that includes a customized br-ex bridge
include::modules/installation-two-node-creating-manifest-custom-br-ex.adoc[leveloffset=+1]

[role="_additional-resources"]
== Additional resources

* xref:../../../installing/installing_bare_metal/ipi/ipi-install-installation-workflow.adoc#creating-manifest-file-customized-br-ex-bridge_ipi-install-installation-workflow[Creating a manifest file for a customized br-ex bridge]

> * For more information about the Red Hat high availability stack, including Pacemaker, Corosync, and fencing concepts, see link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/configuring_and_managing_high_availability_clusters/index[Configuring and managing high availability clusters in RHEL].

Author: Removed the lead-in text.
